samples/dashboards/demos/climate/demo.js

```diff
@@ -127,60 +254,53 @@ async function setupDashboard() {
                 },
                 xAxis: {
                     visible: false,
-                    min: initialMin,
-                    minRange: minRange,
-                    maxRange: maxRange,
+                    min: activeSelection[0],
+                    max: activeSelection[1],
+                    minRange: 30 * 24 * 3600 * 1000, // 30 days
+                    maxRange: 2 * 365 * 24 * 3600 * 1000, // 2 years
                     events: {
-                        afterSetExtremes: async function (e) {
-                            const table =
-                                await dataPool.getConnectorTable(cityScope);
-                            const data = await buildCityData(
-                                e.min || e.target.min,
-                                e.max || e.target.max
-                            );
-                            const lastPoint = data[data.length - 1];
-                            const startIndex =
-                                table.getRowIndexBy('time', data[0][0]);
-
-                            worldDate = lastPoint[0];
-
-                            citiesMap.setData(await buildCitiesMap());
-
-                            // Update only if dragging the navigator
-                            if (e.trigger === 'navigator') {
-                                updateKPI(false);
-                            }
-
-                            if (!cityGrid || !citySeries) {
-                                return; // not ready
-                            }
-
-                            cityGrid.scrollToRow(startIndex);
-                            cityGrid.update(); // Force redraw;
-
-                            citySeries.update({ data });
+                        afterSetExtremes: function (e) {
+                            window.clearTimeout(selectionTimeout);
```

Comment:
IMO the lag between changing extremes is too big. Can we improve that or add some indication that the data is being processed/updated?